### PR TITLE
feat: add lint and unit test workflow checks for pull requests

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- add lint, type check, and unit test checks to CI workflow
	- test snapshots are added to aid in visually inspecting what has changed since we are converting some JSON payload into another JSON type payload that will be sent to Slack
- replace Python map of mixed message payloads with discrete JSON payloads separated by file and by type. JSON event payloads that are in the SNS message format are stored under `functions/messages/`, and JSON raw event message payloads are stored under `functions/events/*`. SNS messages are what are currently trigger the lambda handler, but by having the raw events as well we can make developing and testing custom messages for new event types easier; in the end they are just stringified JSON payloads store din the `Message` attribute of the SNS message that triggers the function
- the `slack-notify-simple` example has been updated to produce an environment variables file to aid in local integrations testing
- an integration testing file `integration_test.py` has been added to test the resources live (using the SNS topic and Lambda handler created by the `slack-notify-simple` example project)
- documentation has been updated to explain and demonstrate how to work with the Python codebase and tests locally

## Motivation and Context
- Improve testing facilities to aid in pull request reviews; next step is to start adding more event types supported

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
	- Unit tests => see GitHub action workflow results or run `pipenv run test`
	- Integration test => deployed `slack-notify-simple` project using my provided webhook URL and then ran `pipenv run python integration_test.py`

### CODE REVIEW

File: functions/notify_slack_test.py [hunk 1]
```diff
@@ -1,204 +1,121 @@
...
+import os
 
 import notify_slack
 import pytest
-from json import loads
-from os import environ
-
-events = (
-    (
-        {
```
Comment (bryantbiggs) [inline functions/notify_slack_test.py:10]: this threw me for a loop. some events in the SNS event format thats delivered to the lambda, and others in the raw message format (that would in theory be passed to something like `aws sns publish --topic-arn "xxx" --message "<message>"`)

File: functions/notify_slack.py [hunk 2]
```diff
@@ -1,161 +1,334 @@
...
+import urllib.parse
+import urllib.request
+from enum import Enum
+from typing import Any, Dict, Optional, Union, cast
+from urllib.error import HTTPError
+
+import boto3
+
+# Set default region if not provided
+REGION = os.environ.get("AWS_REGION", "us-east-1")
```
Comment (antonbabenko) [inline functions/notify_slack.py:23]: What a lovely region :))

File: .gitignore [hunk 3]
```diff
@@ -28,8 +28,51 @@ override.tf.json
 .terraformrc
 terraform.rc
 
-# Lambda directories
-builds/
```
Comment (antonbabenko) [inline .gitignore:32]: `builds/` is used by the Lambda module.
`functions/pytest.ini` (or `.envrc`) may contain secrets.

Probably they worth to be ignored.

Comment (bryantbiggs) [reply .gitignore:32]: image

File: functions/notify_slack.py (hunk 2 above)
Comment (bryantbiggs) [reply functions/notify_slack.py:23]: image

File: .gitignore (hunk 3 above)
Comment (bryantbiggs) [reply .gitignore:32]: re-added in https://github.com/terraform-aws-modules/terraform-aws-notify-slack/pull/152/commits/e091ea237fafd45d39ea5cb627e00877f3c6e55f

**Comment by antonbabenko** [comment]
This PR is included in version 4.22.0 :tada:

**Comment by antonbabenko** [comment]
Very nice! Glad to see some more improvements in this rather popular module. Thanks @bryantbiggs !

**Pull request merged**